### PR TITLE
clarity and modernization after lastdom

### PR DESCRIPTION
**Pull request by vixie**
given we rototilled find_jobs(), it seems like a free throw to add some 'const' and otherwise clarify the invariants.

### CODE REVIEW

**Comment by millert** [comment]
I think you need to rebase this to master, it contains some changes that have already been merged.

**Comment by vixie** [comment]
> I think you need to rebase this to master, it contains some changes that have already been merged.

yikes. merged, since rebase didn't take. try now?

File: cron.c [hunk 1]
```diff
@@ -304,25 +302,27 @@ find_jobs(int vtime, cron_db *db, int doWild, int doNonWild) {
...
 	 */
-	for (u = db->head; u != NULL; u = u->next) {
-		for (e = u->crontab; e != NULL; e = e->next) {
-			bool lastdom = (e->flags & DOM_LAST) != 0 &&
-				tom.tm_mday == 1;
-			bool thisdom = lastdom || bit_test(e->dom, dom) != 0;
-			bool thisdow = bit_test(e->dow, dow);
-
+	const bool is_lastdom = (tom.tm_mday == 1);
+	for (user *u = db->head; u != NULL; u = u->next) {
```
Comment (zhjwpku) [inline cron.c:306]: I think `user *u` can also clarified with const, i.e. `const user *u`.
The same for `entry *e`.

**Review by zhjwpku — CHANGES_REQUESTED** [review]

File: cron.c (hunk 1 above)
Comment (vixie) [reply cron.c]: good catch.

`diff --git a/cron.c b/cron.c
index 49a2005..c3653d6 100644
--- a/cron.c
+++ b/cron.c
@@ -303,8 +303,8 @@ find_jobs(int vtime, cron_db *db, int doWild, int doNonWild) {
         * like many bizarre things, it's the standard.
         */
        const bool is_lastdom = (tom.tm_mday == 1);
-       for (user *u = db->head; u != NULL; u = u->next) {
-               for (entry *e = u->crontab; e != NULL; e = e->next) {
+       for (const user *u = db->head; u != NULL; u = u->next) {
+               for (const entry *e = u->crontab; e != NULL; e = e->next) {
                        Debug(DSCH|DEXT, ("user [%s:%ld:%ld:...] cmd=\"%s\"\n",
                            e->pwd->pw_name, (long)e->pwd->pw_uid,
                            (long)e->pwd->pw_gid, e->cmd))`

**Review by zhjwpku — APPROVED** [review]

**Pull request merged**